### PR TITLE
ci: Add Bitrise workflow to test android app upgrade using appium on Browserstack

### PR DESCRIPTION
**Pull request by Cal-L**
## **Description**

The following changes has been made to `bitrise.yml`

`run_android_appium_test` - New workflow for testing Android appium tests for QA builds. This defaults to run performance tests unless `BROWSERSTACK_TAG_EXPRESSION` is specified. Run this instead of `build_android_qa` if start times are needed.
`run_tag_upgrade_android` - New workflow for running app upgrade e2e test on appium for Android QA builds. Uses `run_android_appium_test` under the hood and sets `BROWSERSTACK_TAG_EXPRESSION`
`build_android_qa` - No longer runs Android performance tests
`release_builds_to_store_pipeline` - Added stage to create QA build alongside RCs. The QA build is later referenced in future `run_tag_upgrade_android` workflows
`release_e2e_pipeline` - Added `run_tag_upgrade_android` to run app upgrade for Android QA builds

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by Cal-L** [comment]
`run_e2e_pipeline` showing upgrade test success - https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/d8d8b90d-20e8-4244-9452-b440c66b208d. The failures on other tests are not related to this branch

File: bitrise.yml [hunk 1]
```diff
@@ -416,6 +418,20 @@ workflows:
...
       - android_e2e_test
+  run_tag_upgrade_android:
+    meta:
+      bitrise.io:
+        stack: linux-docker-android-20.04
+        machine_type_id: elite-xl
+    envs:
+      - PRODUCTION_APP_URL: 'bs://36be83a5e1d585301f833ba0f031f7ecf968d600' # Last production's QA build
+      - PRODUCTION_BUILD_VERSION_NAME: 7.18.0
+      - PRODUCTION_BUILD_VERSION_NUMBER: 1281
```
Comment (SamuelSalas) [inline bitrise.yml:429]: are these variables suppose to be hard coded?

Comment (chrisleewilcox) [reply bitrise.yml:429]: Eventually I think these values will be parameterized but not sure how we want to deal with it for now.  @Cal-L do we want to address these hardcoded values in this PR?

Comment (Cal-L) [reply bitrise.yml:429]: For MVP, we will be leaving it in `bitrise.yml`. We can continue to discuss where to move these in the future async.

**Review by SamuelSalas — APPROVED** [review]
LGTM

**Pull request merged**